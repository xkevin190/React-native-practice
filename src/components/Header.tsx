import * as React from 'react';
import {
  Text,
  View,
  StyleSheet,
  StatusBar,
  TouchableOpacity,
} from 'react-native';
import {NavigationProps} from '../constants/types';
import {normalize, SCREEN_HEIGHT, SCREEN_WIDTH} from '../constants/utils';
import Icon from 'react-native-vector-icons/Ionicons';
import {colors} from '../constants/colors';

type headerProps = {
  navigation?: NavigationProps;
  title: string;
};


const Header = (props: headerProps) => {
  const goToback = () => {
    props.navigation.openDrawer();
  };

  return (
    <View style={styles.container}>
      <StatusBar backgroundColor={colors.background} />
      <TouchableOpacity onPress={goToback}>
        <Icon color={colors.primary} name="menu" size={normalize(23)} />
      </TouchableOpacity>
      <View>
        <Text style={styles.TextHeader}>{props.title}</Text>
      </View>
      <View style={{paddingRight: SCREEN_WIDTH * 0.04}} />
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.background,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: SCREEN_WIDTH * 0.04,
    height: SCREEN_HEIGHT * 0.06,
  },
  TextHeader: {
    fontFamily: 'Urbanist',
    fontSize: normalize(20),
    fontWeight: '600',
    color: colors.primary,
  },
});
