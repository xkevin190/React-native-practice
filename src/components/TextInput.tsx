import * as React from 'react';
import {Text, View, StyleSheet, TextInput} from 'react-native';
import {colors} from '../constants/colors';
import {SCREEN_HEIGHT} from '../constants/utils';

interface InputProps {
  onchange: (text: string) => void;
}

const Input = (props: InputProps) => {
  return (
    <TextInput
      style={styles.container}
      onChangeText={text => props.onchange(text)}
      placeholder={'writte here'}
    />
  );
};

export default Input;

const styles = StyleSheet.create({
  container: {
    marginTop: SCREEN_HEIGHT * 0.03,
    height: SCREEN_HEIGHT * 0.055,
    marginHorizontal: 20,
    borderRadius: 14,
    borderColor: colors.primary,
    paddingHorizontal: 20,
    borderWidth: 0.8,
    fontSize: 15,
    color: colors.primary,
  },
});
