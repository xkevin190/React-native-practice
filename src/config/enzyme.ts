import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

/**
 * @tech-debt
 *
 * Update enzyme to support react 18
 */
Enzyme.configure({ adapter: new Adapter() });
