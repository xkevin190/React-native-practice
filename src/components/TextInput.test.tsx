import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import Header from './Header';
import TextInput from './TextInput';

describe('TextInput (Unit-Test)', () => {
  let wrapper: ShallowWrapper;

  const onChange = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<TextInput onchange={onChange} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('onchange should executed', () => {
    wrapper.find('TextInput').first().simulate('changeText', 'test');
    expect(onChange).toBeCalledWith('test');
  });
});
