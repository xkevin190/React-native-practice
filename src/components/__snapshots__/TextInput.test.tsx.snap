// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput (Unit-Test) should match snapshot 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder="writte here"
  style={
    Object {
      "borderColor": "#ffffff",
      "borderRadius": 14,
      "borderWidth": 0.8,
      "color": "#ffffff",
      "fontSize": 15,
      "height": 73.37,
      "marginHorizontal": 20,
      "marginTop": 40.019999999999996,
      "paddingHorizontal": 20,
    }
  }
/>
`;
